### PR TITLE
kornia-apriltag: Allow configuration of max hamming distance for detections

### PR DESCRIPTION
**Pull request by rossng**
## 📝 Description

**Fixes/Relates to:** #450

Succeeds #620 

---

## 🛠️ Changes Made
- [ ] Make max hamming distance for AprilTag code matching configurable through `QuickDecode::set_max_hamming` and `TagFamily::with_max_hamming`.
- [ ] Re-enable AprilTag detection tests on aarch64

---

## 🧪 How Was This Tested?
- [x] **Unit Tests:** `test_configurable_max_hamming` tries a few different permutations of code and detector config.
- [ ] **Manual Verification:** No manual verification performed
- [ ] **Performance/Edge Cases:** N/A

---

## 🕵️ AI Usage Disclosure
*Check one of the following:*
- [ ] 🟢 **No AI used.**
- [x] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.
- [ ] 🔴 **AI-generated:** (Note: These PRs may be subject to stricter scrutiny or immediate closure if the logic is not explained).

---

## 🚦 Checklist
- [ ] I am assigned to the linked issue (required before PR submission)
- [ ] The linked issue has been approved by a maintainer
- [ ] This PR strictly implements what the linked issue describes (no scope creep)
- [ ] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [ ] My code follows the existing style guidelines of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] (Optional) I have attached screenshots/recordings for UI changes.

---

## 💭 Additional Context
Add any other context or screenshots about the pull request here.

### CODE REVIEW

**Comment by rossng** [comment]
I agree but suggest we tackle this in a separate PR

**Comment by sidd-27** [comment]
Actually nvm if we reduce the number of buckets the size of each bucket increases drastically, 4 seems to be the sweet spot

**Comment by sidd-27** [comment]
@rossng  could you fix the lint errors? also add errors or warnings for families like 16h5 where a detection of 3 bit could give wrong results

**Comment by rossng** [comment]
I'll address these today

**Comment by rossng** [comment]
I've added a `min_hamming` field to `TagFamily` so that we can calculate what the maximum safe hamming distance for each family is.

There's a new `MaxHammingTooLarge` to represent the error, distinct from the `InvalidAllowedErrors` (I think there's a possibility of eliminating the latter in the future, as it would be possible to e.g. dynamically switch QuickDecode implementations to support max_hamming > 3).

Also rewrote the commit history so each commit has been cargo fmt-ed.

File: kornia-py/src/apriltag.rs [hunk 1]
```diff
@@ -392,10 +392,25 @@ pub mod family {
...
+        pub fn get_max_hamming(&self) -> u8 {
+            self.0.max_hamming()
+        }
+
+        #[setter]
+        pub fn set_max_hamming(&mut self, value: u8) -> PyResult<()> {
+            self.0
+                .set_max_hamming(value)
+                .map_err(|e| PyErr::new::<PyException, _>(e.to_string()))
         }
```
[the remark this answers is not in the document]
Comment (rossng) [reply kornia-py/src/apriltag.rs:418]: added python tests and ran them with pixi

**Comment by sidd-27** [comment]
LGTM

**Comment by rossng** [comment]
Feel free to merge if you're happy with it. Not sure if it's only @edgarriba that can do this?

**Comment by rossng** [comment]
I just checked and it rebases cleanly to master. Will fix anything that comes up when CI runs.

**Comment by sidd-27** [comment]
the CI tests seem to be broken, i dont think anythings is wrong on your end tho, looks good to merge after CI is fixed

**Comment by sidd-27** [comment]
CI tests seem to pass, i think its just the sscache thing that's failing, merging this

**Review by sidd-27 — APPROVED** [review]
